application/org.openjdk.jmc.flightrecorder.flameview/src/main/java/org/openjdk/jmc/flightrecorder/flameview/tree/TraceTreeUtils.java

```diff
@@ -43,12 +44,34 @@
 import org.openjdk.jmc.flightrecorder.stacktrace.StacktraceModel.Fork;
 import org.openjdk.jmc.flightrecorder.stacktrace.FrameSeparator.FrameCategorization;
 import org.openjdk.jmc.flightrecorder.stacktrace.StacktraceFrame;
+import org.openjdk.jmc.flightrecorder.stacktrace.messages.internal.Messages;
 
 public class TraceTreeUtils {
 	public final static String DEFAULT_ROOT_NAME = "__root";
 	public final static String DEFAULT_ROOT_PACKAGE_NAME = "";
 	public final static FrameSeparator DEFAULT_FRAME_SEPARATOR = new FrameSeparator(FrameCategorization.METHOD, false);
 
+	private static class TraceNodeFactory {
+
+		private static TraceNode getRootTraceNode(String rootName, Fork rootFork) {
```

Comment:
Is this really necessary? The factory class doesn't add much in terms of abstraction over having a factory method, and it's not clear to me why the get by frame method belongs in there and not the rest of the utility methods. I'd revert these changes, or at least add the helper methods as private statics without the extra inner class.

Reply:
agree, helper methods are fine.